lib/Support/IRHelpers.cpp

```diff
@@ -78,6 +78,7 @@ static std::string mangleName(StringRef String) {
 }
 
 Constant *getUniqueString(Module *M, StringRef String, StringRef Namespace) {
+  revng_assert(not Namespace.empty());
```

Comment:
I'm OK with this, maybe add docs. But frankly I was thinking about dropping `Namespace` altogether and have a fixed one.

I suppose you triggered this because you're explicitly specifying it (as opposed to using the default value for this argument). Do you *really* care about having a distinct global variable if there's another user of `getUniqueString` requesting the same string? Do you need them to have different addresses? Do you need to change its content?

If not, let's collapse them all in a single namespace. If you're serializing `MetaAddress` or a `QualifiedType` or the like there's a good chance we get deduplication for free and save a lot of space.

Reply:
We're using this for string literals to be emitted in C, so no deduplication for `MetaAddress` and `QualifiedType`. On the other hand I expect potential deduplication due to the same string literal being used possibly in many places in C.

For what concerns namespaces, I don't really care about them though. For what I am concerned we can just get rid of the `Namespace` argument alltogether, and only always use the same namespace.

Reply:
I've some FUD that one day we'll need to distinguish strings, let's this feature survive for a little longer.
In the meantime, let's use the default value instead of explicitly choosing a namespace.

Reply:
We already have default value for this, so I think for now it's resolved.

Reply:
> We already have default value for this

Not sure what you mean.

I'm suggesting to drop the manually specified namespace name in `revng-c` when invoking `getUniqueString`.

Reply:
> I'm suggesting to drop the manually specified namespace name in revng-c when invoking getUniqueString.

@kolodziej you can do this. I'm sure it doesn't affect any other part of your PR in revng-c

Reply:
Sure

Reply:
Ok so this is solved. I'm closing this conversation.